frontend/zulu/src/components/Map/MapGeojsonMarkers.jsx

```diff
@@ -46,28 +50,32 @@ class MapComponent extends React.Component {
         console.log(coords)
     };
 
+    setLocationState() {
 
-    componentDidMount() {
-
-        const geo = navigator.geolocation;
+         navigator.geolocation.getCurrentPosition((position) => {
+            this.setState({lat: position.coords.latitude, lng: position.coords.longitude})
+          },
+           (error) => {},
+           {enableHighAccuracy: true, timeout: 20000, maximumAge: 10000}
+          );
 
-        const watcher = geo.watchPosition(this.onChange, this.onError);
     }
+    componentDidMount() {
+        
+        this.setLocationState();
 
+        // TODO: figure out why this causes all points at GeoJsonLayer
+        // component to re-render
+        this.selfLocationInterval = setInterval(() => {
+                                this.setLocationState(); 
+                                }, 
+                                15000);
+      };
 
     render() {
         var center = [this.state.lat, this.state.lng];
+        var closePointsURL = "http://localhost:8342/api/point?longitude=" + this.state.lng + "&latitude=" + this.state.lat + "&max_distance=5000";
```

Comment:
might be easier to change params on the fly if this was wrapped in a method

Reply:
yhe - 100% agree, more of "hack" I did than a real solution.

This would need to be refactored anyways:
1. We'll need different endpoint for dev/prod
2. The max_distance parameter should somehow be controlled by your zoom, like if you're zooming on area which is equivalent to 100KM, max_distance should be updated accordingly.
3. Will need to add authentication for the user id layer. 

So I prefer we do it later (probably with the user authentication layer).